### PR TITLE
DEV-1679: Add unique_award_key to transaction records

### PR DESCRIPTION
**Pull request by michaeldhess**
**High level description:**
We want to start looking into relationships between Transactions and Awards on the Broker, so we are adding a `unique_award_key` to FPDS and FABS (including historical) transactions.

**Technical details:**
Added `unique_award_key` to the DetachedAwardProcurement, DetachedAwardFinancialAssistance, and PublishedAwardFinancialAssistance tables. Updated FPDS loader script and FABS validations to derive and populate this field within their records. Created a script to backfill all historical records.

**Link to JIRA Ticket:**
[DEV-1679](https://federal-spending-transparency.atlassian.net/browse/DEV-1679)

The following are ALL required for the PR to be merged:
- [ ] Backend review completed
- Merged concurrently with Frontend N/A
- Unit & integration tests updated with relevant test cases N/A (not currently used in the application)
- Frontend impact assessment completed N/A

### CODE REVIEW

File: dataactcore/scripts/pullFPDSData.py [hunk 1]
```diff
@@ -2259,6 +2261,16 @@ def create_unique_key(row):
...
 
 
+def create_unique_award_key(row):
+    key_list = ['piid', 'agencyid', 'idvpiid', 'idvagencyid'] if row['pulled_from'] == 'award' else ['piid', 'agencyid']
+    unique_string_list = [] if row['pulled_from'] == 'award' else [row['pulled_from']]
+
+    for item in key_list:
+        unique_string_list.append(row[item] if row[item] and str(row[item]) != 'nan' else '-none-')
+
+    return '_'.join(unique_string_list)
```
Comment (alburde1) [inline dataactcore/scripts/pullFPDSData.py]: This is only handling the key for historical data. What about for new data coming in?

File: dataactvalidator/scripts/loadTas.py [hunk 2]
```diff
@@ -147,6 +147,7 @@ def load_tas(backfill_historic=False):
             backfill_historic: if set to true, this will only update certain columns if budget_function_code is null
     """
     # read TAS file to dataframe, to make sure all is well with the file before firing up a db transaction
+    sess = GlobalDB.db().session
```
Comment (alburde1) [inline dataactvalidator/scripts/loadTas.py]: Why are you creating this a second time in the same function?

File: dataactvalidator/validation_handlers/validationManager.py [hunk 3]
```diff
@@ -312,10 +312,17 @@ def run_validation(self, job):
...
                     else:
                         if file_type == "fabs":
+                            # Create afa_generated_unique
                             record['afa_generated_unique'] = (record['award_modification_amendme'] or '-none-') + "_" +\
                                                              (record['awarding_sub_tier_agency_c'] or '-none-') + \
                                                              "_" + (record['fain'] or '-none-') + "_" + \
                                                              (record['uri'] or '-none-')
+                            # Create unique_award_key
+                            identifier = record['fain'] if str(record['record_type']) == '1' else record['uri']
+                            record['unique_award_key'] = (record['record_type'] or '-none-') + '_' + \
```
Comment (alburde1) [inline dataactvalidator/validation_handlers/validationManager.py]: The `record_type` here would be 1-3, in the script it's `NON` or `AGG`, we should pick one.

File: dataactcore/scripts/populate_historical_unique_award_keys.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        logger.info('Populating unique_award_key for unpublished FPDS non-aggregated records...')
+        sess.query(dafa).filter(dafa.record_type != '1').\
+            update({dafa.unique_award_key: func.concat('NON_', func.coalesce(dafa.fain, '-none-'), '_',
+                                                       func.coalesce(dafa.awarding_sub_tier_agency_c, '-none-'))},
+                   synchronize_session=False)
+        logger.info('Unpublished FABS non-aggregated records populated.\n')
+
+        logger.info('Populating unique_award_key for published FABS aggregate records...')
+        pafa = PublishedAwardFinancialAssistance
+        sess.query(pafa).filter(pafa.record_type == '1').\
```
Comment (alburde1) [inline dataactcore/scripts/populate_historical_unique_award_keys.py]: record_type in PublishedAwardFinancialAssistance and DetachedAwardFinancialAssistance are integers, is this going to work this way?

File: dataactcore/scripts/populate_historical_unique_award_keys.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+            update({pafa.unique_award_key: func.concat('AGG_', func.coalesce(pafa.uri, '-none-'), '_',
+                                                       func.coalesce(pafa.awarding_sub_tier_agency_c, '-none-'))},
+                   synchronize_session=False)
+        logger.info('Published FABS aggregate records populated.\n')
+
+        logger.info('Populating unique_award_key for published FPDS non-aggregated records...')
```
Comment (alburde1) [inline dataactcore/scripts/populate_historical_unique_award_keys.py]: This should say `FABS`

File: dataactcore/scripts/populate_historical_unique_award_keys.py [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+        logger.info('Populating unique_award_key for unpublished FABS aggregate records...')
+        dafa = DetachedAwardFinancialAssistance
+        sess.query(dafa).filter(dafa.record_type == '1').\
+            update({dafa.unique_award_key: func.concat('AGG_', func.coalesce(dafa.uri, '-none-'), '_',
+                                                       func.coalesce(dafa.awarding_sub_tier_agency_c, '-none-'))},
+                   synchronize_session=False)
+        logger.info('Unpublished FABS aggregate records populated.\n')
+
+        logger.info('Populating unique_award_key for unpublished FPDS non-aggregated records...')
```
Comment (alburde1) [inline dataactcore/scripts/populate_historical_unique_award_keys.py]: This should say `FABS`

File: dataactcore/scripts/populate_historical_unique_award_keys.py (hunk 4 above)
Comment (michaeldhess) [reply dataactcore/scripts/populate_historical_unique_award_keys.py]:
```SELECT *
FROM published_award_financial_assistance
WHERE record_type = 1 AND record_type = '1'
LIMIT 1;```
Returns values, which means `sess.query(pafa).filter(pafa.record_type == '1')` could use string or int and it’ll work either way.

**Pull request merged**